From 49c9ab7ddd8c20ad1722d20185a0fc3e71ce2d20 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 26 May 2010 14:24:15 +0000
Subject: Don't fail to browse the device if already mounted

We don't need to have an error if the device is already mounted,
just show it already!
---
diff --git a/applet/main.c b/applet/main.c
index 5655a75..dd517c8 100644
--- a/applet/main.c
+++ b/applet/main.c
@@ -114,9 +114,16 @@ mount_finish_cb (GObject *source_object,
 
 	if (g_file_mount_enclosing_volume_finish (G_FILE (source_object),
 						  res, &error) == FALSE) {
-		g_printerr ("Failed to mount OBEX volume: %s", error->message);
-		g_error_free (error);
-		return;
+		/* Ignore "already mounted" error */
+		if (error->domain == G_IO_ERROR &&
+		    error->code == G_IO_ERROR_ALREADY_MOUNTED) {
+			g_error_free (error);
+			error = NULL;
+		} else {
+			g_printerr ("Failed to mount OBEX volume: %s", error->message);
+			g_error_free (error);
+			return;
+		}
 	}
 
 	uri = g_file_get_uri (G_FILE (source_object));
--
cgit v0.8.3.1
